receiver/receiverhelper/obsreport_test.go

```diff
@@ -296,10 +296,30 @@ func TestCheckReceiverLogsViews(t *testing.T) {
 	assert.Error(t, tt.CheckReceiverLogs(transport, 0, 7))
 }
 
+type myError struct {
+	error
+}
+
+func returnsError(b bool) error {
+	var p *myError = nil
+	if b {
+		p = &myError{error: errors.New("my error")}
+	}
+	return p // Will always return a non-nil error.
+}
+
+func TestIsErrorNil(t *testing.T) {
+	assert.True(t, isErrorNil(nil))
+	assert.False(t, isErrorNil(returnsError(true)))
+	assert.False(t, returnsError(true) == nil)
+	assert.True(t, isErrorNil(returnsError(false)))
+	assert.False(t, returnsError(false) == nil)
```

Comment:
Linter is complaining about these two lines, I guess the best way of putting it is this?
```suggestion
	assert.NotNil(t, returnsError(true))
	assert.True(t, isErrorNil(returnsError(false)))
	assert.NotNil(t, returnsError(false))
```

Reply:
Unfortunately `assert.NotNil` (`assert.Nil`) does the right checks, so will not work

```golang
// isNil checks if a specified object is nil or not, without Failing.
func isNil(object interface{}) bool {
	if object == nil {
		return true
	}

	value := reflect.ValueOf(object)
	switch value.Kind() {
	case
		reflect.Chan, reflect.Func,
		reflect.Interface, reflect.Map,
		reflect.Ptr, reflect.Slice, reflect.UnsafePointer:

		return value.IsNil()
	}

	return false
}
```